map/bookmark_manager.cpp

```diff
@@ -300,25 +300,37 @@ Bookmark * BookmarkManager::CreateBookmark(kml::BookmarkData && bmData)
   return AddBookmark(std::make_unique<Bookmark>(std::move(bmData)));
 }
 
-Bookmark * BookmarkManager::CreateBookmark(kml::BookmarkData && bm, kml::MarkGroupId groupId)
+Bookmark * BookmarkManager::CreateBookmark(kml::BookmarkData && bmData, kml::MarkGroupId groupId)
 {
   CHECK_THREAD_CHECKER(m_threadChecker, ());
 
   auto const & c = classif();
   CHECK(c.HasTypesMapping(), ());
   std::stringstream ss;
-  for (size_t i = 0; i < bm.m_featureTypes.size(); ++i)
+  for (size_t i = 0; i < bmData.m_featureTypes.size(); ++i)
   {
-    ss << c.GetReadableObjectName(c.GetTypeForIndex(bm.m_featureTypes[i]));
-    if (i + 1 < bm.m_featureTypes.size())
+    ss << c.GetReadableObjectName(c.GetTypeForIndex(bmData.m_featureTypes[i]));
+    if (i + 1 < bmData.m_featureTypes.size())
       ss << ",";
   }
 
-  bm.m_timestamp = kml::TimestampClock::now();
-  bm.m_viewportScale = static_cast<uint8_t>(df::GetZoomLevel(m_viewport.GetScale()));
+  bmData.m_timestamp = kml::TimestampClock::now();
+  bmData.m_viewportScale = static_cast<uint8_t>(df::GetZoomLevel(m_viewport.GetScale()));
 
-  auto * bookmark = CreateBookmark(std::move(bm));
+  // Recover the bookmark from the recently deleted.
+  // The recently deleted bookmark exists only when deleted from the Place Page screen.
+  if (HasRecentlyDeletedBookmark())
```

Comment:
No.
Because the `bmData` we get from the method argument
```
Bookmark * BookmarkManager::CreateBookmark(kml::BookmarkData && bmData, kml::MarkGroupId groupId)
{
```
 and it is the data but not the bookmark itself. `bmData` is used in the constructor to initialize the Bookmark.
`auto bookmark = CreateBookmark(std::move(bmData));`

In `if (HasRecentlyDeletedBookmark())` scope we replace the `data` with the recently deleted (recently deleted bookmark data, not bookmark itself).